### PR TITLE
feat(lib/xchain): use commitment level

### PR DESCRIPTION
**Pull request by Lazar955**
Use commitment level when fetching blocks. Extends netconf and modifies xprovider

task: none

### CODE REVIEW

File: lib/netconf/netconf.go [hunk 1]
```diff
@@ -77,14 +77,15 @@ func (n Network) Chain(id uint64) (Chain, bool) {
...
-	BlockPeriod   time.Duration // Block period of the chain
+	ID              uint64        // Chain ID asa per https://chainlist.org
+	Name            string        // Chain name as per https://chainlist.org
+	RPCURL          string        // RPC URL of the chain
+	AuthRPCURL      string        // RPC URL of the chain with JWT authentication enabled
+	PortalAddress   string        // Address of the omni portal contract on the chain
+	DeployHeight    uint64        // Height that the portal contracts were deployed
+	IsOmni          bool          // Whether this is the Omni chain
+	BlockPeriod     time.Duration // Block period of the chain
+	CommitmentLevel string        // Commitment Level of the block ("latest", "finalized")
```
Comment (corverroos) [inline lib/netconf/netconf.go:88]: "latest" "finalised" might not be enough for rollups like ZKrollups where we will need a completely different strategy.

So rather name this `FinalizationStrat`.

Also please make this an `string enum` in netconf with two values `StratLatest` and `StratFinalized`

File: test/e2e/types/chain.go [hunk 2]
```diff
@@ -9,23 +9,26 @@ import (
 //nolint:gochecknoglobals // Static mappings
 var (
 	ChainOmniEVM = EVMChain{
-		Name:        "omni_evm",
-		ID:          1,
-		BlockPeriod: time.Second,
+		Name:            "omni_evm",
+		ID:              1,
+		BlockPeriod:     time.Second,
+		CommitmentLevel: "finalized",
```
Comment (corverroos) [inline test/e2e/types/chain.go:15]: use netconf constants here

File: test/e2e/types/chain.go [hunk 3]
```diff
@@ -36,9 +39,10 @@ func AnvilChainsByNames(names []string) []EVMChain {
 	var chains []EVMChain
 	for i, name := range names {
 		chains = append(chains, EVMChain{
-			Name:        name,
-			ID:          anvilChainIDFactor * uint64(i+1),
-			BlockPeriod: time.Second,
+			Name:            name,
+			ID:              anvilChainIDFactor * uint64(i+1),
+			BlockPeriod:     time.Second,
+			CommitmentLevel: "latest",
```
Comment (corverroos) [inline test/e2e/types/chain.go:45]: add a comment why latest here

File: lib/xchain/provider/fetch.go [hunk 4]
```diff
@@ -15,26 +16,18 @@ import (
...
-	// params := []string{"latest", "false"}
-	// err := rpcClient.Client().CallContext(ctx, &finalisedHeader, "eth_getBlockByNumber", params)
-	// if err != nil {
-	//	 return nil, errors.Wrap(err, "could not get finalized block")
-	// }
-
-	// TODO(corver): Support different finalized methods (to be added to netconf).
-	//  The only chain we support at this point is anvil, it doesn't support "finalized", so just use "latest" for now.
-	height, err := rpcClient.BlockNumber(ctx)
+func getCurrentFinalisedBlockHeader(
```
Comment (pavelnikolov) [inline lib/xchain/provider/fetch.go:19]: Please, [don't use `get` as function or method prefixes](https://google.github.io/styleguide/go/decisions#getters). It is neither go-idiomatic, nor necessary.

**Pull request merged**